grafana/grafana
# Elasticsearch: Implement processing of logs query results in backend

Part of https://github.com/grafana/grafana/issues/54011. This PR adds processing of logs response to backend. To do this, we have encapsulated all logic into `processLogResponse`.

`processLogResponse` processes response in the same way as it is currently done on frontend with 2 exceptions:
- `highlight` needs to be implemented. I'll create separate PR for that.
- `level` field on frontend was added to the third position (after timestamp and line), but this is not needed, as we detect it based on name of the field and not its position. This meant, that in log details, level field wasn't alphabetically sorted. As there is no dependency on position of `level` I decided to keep it's position based on alphabetical order. 

I encapsulated logic that was same for `raw_data` and `logs` to `processDocsToDataFrameFields`.  At one point, I tried to encapsulate logic for `flattening` and creating `propNames` and `docs` into 1 function, but there are slight differences and I have decided against it. Moreover, in next PR, I need to remove creating `_source` for `raw_data` response which will make creating `docs` and `propNames` even more different.

For processing of log result, we needed access to `logMessageField` and `logLevelField` from data source config page. Therefore I have grouped all `configuredFields` into struct and passed them to response processing function.

**Special notes for your reviewer**:

To test this
1. Run ElasticSearch data source by `make devenv sources=elastic`
2. Use `elasticsearchBackendMigration` feature toggle to toggle if query is run trough frontend/backend
3. Run `logs` query
4. Results run trough frontend and backend should be the same (with exception of highlighting and position of level field)
